cpp/src/DataStorm/NodeSessionManager.cpp

```diff
@@ -119,18 +133,18 @@ NodeSessionManager::createOrGet(
     _sessions.emplace(node->ice_getIdentity(), session);
 
     instance->getConnectionManager()->add(
-        node,
+        make_shared<NodePrx>(*node),
```

Comment:
It's fine for now, but we should revisit this code later.

Reply:
Added a TODO